src/nurse/tests/test_views.py

```diff
@@ -784,3 +784,26 @@ def test_post_only_staff(self, user, client):
         assert response.json() == {
             "detail": "You do not have permission to perform this action."
         }
+
+
+@pytest.mark.django_db
+class TestUserOneSignalProfileView:
+    url = reverse_lazy("useronesignalprofile-list")
+
+    def test_endpoint_onesignal(self):
+        assert self.url == "/onesignal/"
+
+    def test_create_onesignal(self, client, user):
+        data = {"user": user.id, "subscription_id": "1233456789"}
+        assert UserOneSignalProfile.objects.count() == 0
+        response = client.post(self.url, data, format="json")
+        assert response.status_code == status.HTTP_201_CREATED
+        assert UserOneSignalProfile.objects.count() == 1
+
+    def test_read_onesignal(self, client, user):
+        profile = UserOneSignalProfile.objects.create(
+            user=user, subscription_id="123456789"
+        )
+        response = client.get(self.url, kwargs={"pk": profile.id})
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data[0]["subscription_id"] == "123456789"
```

Comment:
I'm think the read view shouldn't require to pass a pk since we would return the only UserOneSignalProfile that we have for that request.user. So the view should probably override the `get()` method and retrieve the one signal profile using request.user